core/dbt/contracts/state.py

```diff
@@ -1,30 +1,48 @@
 from pathlib import Path
 from .graph.manifest import WritableManifest
 from .results import RunResultsArtifact
+from .results import FreshnessExecutionResultArtifact
 from typing import Optional
 from dbt.exceptions import IncompatibleSchemaException
 
 
 class PreviousState:
-    def __init__(self, path: Path):
+    def __init__(self, path: Path, current_path: Path):
         self.path: Path = path
+        self.current_path: Path = current_path
         self.manifest: Optional[WritableManifest] = None
         self.results: Optional[RunResultsArtifact] = None
+        self.sources: Optional[FreshnessExecutionResultArtifact] = None
+        self.sources_current: Optional[FreshnessExecutionResultArtifact] = None
 
-        manifest_path = self.path / "manifest.json"
+        manifest_path = self.path / 'manifest.json'
         if manifest_path.exists() and manifest_path.is_file():
             try:
-                # we want to bail with an error if schema versions don't match
-                self.manifest = WritableManifest.read_and_check_versions(str(manifest_path))
+                self.manifest = WritableManifest.read(str(manifest_path))
```

Comment:
We want this to use `read_and_check_versions` (instead of `read`), to raise a better exception when the version mismatch is preempted by a contract mismatch. See https://github.com/dbt-labs/dbt-core/pull/4497.

If you swap out all instances of `read` for `read_and_check_versions`, that will get `test_postgres_previous_version_manifest` passing again.

Reply:
Done!